apiserver/src/main/java/org/dependencytrack/resources/v1/PermissionResource.java

```diff
@@ -165,25 +170,37 @@ public Response removePermissionFromUser(
             @Parameter(description = "A valid username", required = true)
             @PathParam("username") String username,
             @Parameter(description = "A valid permission", required = true)
+            @QueryParam("userType") String type,
             @PathParam("permission") String permissionName) {
         try (QueryManager qm = new QueryManager()) {
             return qm.callInTransaction(() -> {
-                User principal = qm.getUser(username);
-                if (principal == null) {
+                User user = qm.getUser(username, (Class<? extends User>) switch (StringUtils.defaultString(type).toLowerCase()) {
+                    case "managed" -> ManagedUser.class;
+                    case "ldap" -> LdapUser.class;
+                    case "oidc" -> OidcUser.class;
+                    default -> User.class;
+                });
```

Comment:
The concrete user type seems to be irrelevant for what this method is doing.

Does this pre-date [ADR-006](https://github.com/DependencyTrack/hyades/blob/2f423efb8478454a54866f15e60672bc3975b251/docs/architecture/decisions/006-consolidate-user-tables.md) per chance, and thus had to deal with the same username existing for multiple user types? If so, I think this can be removed since uniqueness of usernames is now enforced across all types.